src/solvers/NelderMead.jl

```diff
@@ -80,62 +227,75 @@ function NelderMead!(
         retraction_method=retraction_method,
         inverse_retraction_method=inverse_retraction_method,
     )
-    o = decorate_options(o; kwargs...)
-    return get_solver_return(solve(p, o))
+    s = decorate_state(s; kwargs...)
+    solve!(mp, s)
+    return get_solver_return(s)
 end
 #
 # Solver functions
 #
-function initialize_solver!(p::CostProblem, o::NelderMeadOptions)
-    # init cost and x
-    o.costs = get_cost.(Ref(p), o.population)
-    return o.x = o.population[argmin(o.costs)] # select min
+function initialize_solver!(mp::AbstractManoptProblem, s::NelderMeadState)
+    # init cost and p
+    s.costs = get_cost.(Ref(mp), s.population.pts)
+    return s.p = s.population.pts[argmin(s.costs)] # select min
 end
-function step_solver!(p::CostProblem, o::NelderMeadOptions, ::Any)
-    m = mean(p.M, o.population)
-    ind = sortperm(o.costs) # reordering for cost and p, i.e. minimizer is at ind[1]
-    ξ = inverse_retract(p.M, m, o.population[last(ind)], o.inverse_retraction_method)
+function step_solver!(mp::AbstractManoptProblem, s::NelderMeadState, ::Any)
+    M = get_manifold(mp)
+    m = mean(M, s.population.pts)
+    ind = sortperm(s.costs) # reordering for cost and p, i.e. minimizer is at ind[1]
+    ξ = inverse_retract(M, m, s.population.pts[last(ind)], s.inverse_retraction_method)
     # reflect last
-    xr = retract(p.M, m, -o.α * ξ, o.retraction_method)
-    Costr = get_cost(p, xr)
+    xr = retract(M, m, -s.α * ξ, s.retraction_method)
+    Costr = get_cost(mp, xr)
+    continue_steps = true
     # is it better than the worst but not better than the best?
-    if Costr >= o.costs[first(ind)] && Costr < o.costs[last(ind)]
+    if Costr >= s.costs[first(ind)] && Costr < s.costs[last(ind)]
         # store as last
-        o.population[last(ind)] = xr
-        o.costs[last(ind)] = Costr
+        s.population.pts[last(ind)] = xr
+        s.costs[last(ind)] = Costr
+        continue_steps = false
     end
     # --- Expansion ---
-    if Costr < o.costs[first(ind)] # reflected is better than fist -> expand
-        xe = retract(p.M, m, -o.γ * o.α * ξ, o.retraction_method)
-        Coste = get_cost(p, xe)
+    if Costr < s.costs[first(ind)] # reflected is better than fist -> expand
+        xe = retract(M, m, -s.γ * s.α * ξ, s.retraction_method)
+        Coste = get_cost(mp, xe)
         # successful? use the expanded, otherwise still use xr
-        o.population[last(ind)] .= Coste < Costr ? xe : xr
-        o.costs[last(ind)] = min(Coste, Costr)
+        s.population.pts[last(ind)] .= Coste < Costr ? xe : xr
```

Comment:
Maybe we should use `copyto!` here?

Reply:
Or maybe just `s.population.pts[last(ind)] = Coste < Costr ? xe : xr`?

Reply:
see below, I think a copyto would be fine as well, if we want to avoid the allocation.

Reply:
`s.population.pts[last(ind)] = Coste < Costr ? xe : xr` does not allocate. It changes what is stored in the vector `pts`. We would need `copyto!` if we had a cache for intermediate points but currently we don't.

Reply:
ah ok, then let's do that.